com.ibm.streamsx.hdfs/impl/java/src/com/ibm/streamsx/hdfs/HDFS2FileSink.java

```diff
@@ -986,7 +1053,7 @@ public void reset(Checkpoint checkpoint) throws Exception {
 		
 		// close current file
 		if (fFileToWrite != null)
-			closeFile();
+			fFileToWrite.close();
```

Comment:
Why close the file directly instead of calling closeFile()? One of the tasks of closeFile() is to interrupt the fileTimerThread and set it to null. With this change, reset() may leave the thread running in the background. I am concerned that this may cause unusual problems with file expiration once processing continues.